gluten-ut/spark32/src/test/scala/org/apache/spark/sql/GlutenDataFrameSuite.scala

```diff
@@ -297,6 +298,17 @@ class GlutenDataFrameSuite extends DataFrameSuite with GlutenSQLTestsTrait {
     }
   }
 
+  test(GLUTEN_TEST +
+      "Allow leading/trailing whitespace in string for casting to integral type") {
+    Seq(" 123", "123 ", " 123 ", "123\n\n\n", "123\r\r\r", "123\f\f\f", "123\u000C")
+        .toDF("col1").createOrReplaceTempView("t1")
+    val df = spark.sql("select cast(col1 as int) from t1")
+    checkAnswer(df, Row(123) :: Row(123) ::
+        Row(123) :: Row(123) :: Row(123) :: Row(123) :: Row(123) :: Nil)
+    assert(find(df.queryExecution.executedPlan)(
+      _.isInstanceOf[ProjectExecTransformer]).isDefined)
```

Comment:
nit: may be add some test for binary and double type?

Reply:
Yes, just added some more tests. Thanks!